lib/src/group_changes.dart

```diff
@@ -127,23 +151,27 @@ List<SummaryData> summarizePageData(
   final Map<String, int> hashIndex = Map.fromEntries(
       Iterable.generate(hashes.length, (i) => MapEntry(hashes[i], i)));
 
+  // New tests don't have a previous commit or result.
+  // for our heuristic, we group them together if they are within
+  // 10 commits of each other.
```

Comment:
Well, we know when the previous build was (if this isn't a new builder), so we still have the same lower bound available. Not on that record, of course, but we could just look at the other records for the builder. Alternative the script setting `previous_commit_hash` could also set it for new tests by looking at the previous commit on the other tests. Then this code doesn't need any special treatment at all.